src/features/events.rs

```diff
@@ -138,3 +158,86 @@ pub async fn check_events_loop(http: Arc<http::Http>) {
     thread::sleep(time::Duration::from_secs(SLEEP_TIME_SECS))
   }
 }
+
+pub async fn snooze_reaction(ctx: &Context, reaction: &Reaction) {
+  let message = reaction.message(&ctx).await.unwrap();
+  let reaction_author = reaction.user(&ctx).await.unwrap().id;
+  let message_author = message.author.id;
+  let channel = message.channel_id;
+  let content = message.content.clone();
+  if content.starts_with(&format!("<@{}>", reaction_author.0))
+    && message_author.0 == ctx.cache.current_user_id().0
+  {
+    {
+      let mut snooze = SNOOZE.lock().unwrap();
+      if snooze.iter().cloned().any(|s| s.message_id == message.id.0) {
+        return;
+      }
+      snooze.push(RemindMeSnooze {
+        author: reaction_author.0,
+        channel: channel.0,
+        message_id: message.id.0,
+        content: content.replace(format!("<@{}>", reaction_author.0).as_str(), ""),
+      });
+    }
+    let message = channel
+      .say(&ctx.http, "Enter the new duration please")
+      .await
+      .unwrap();
+    SNOOZE_MESSAGE.lock().unwrap().push(message);
+  }
+}
+
+pub async fn snooze_message(ctx: &Context, message: &Message) {
+  let message_author = message.author.id;
+  let channel = message.channel_id;
+  let content = message.content.clone();
+  let list_snooze = SNOOZE.lock().unwrap().clone();
+  if let Some(snooze) = list_snooze
+    .iter()
+    .find(|s| s.channel == channel.0 && s.author == message_author.0)
+  {
+    if let Some(captures) = TIME_INPUT_REGEX.captures(&content) {
+      let trigger_date: Option<DateTime<_>> = extract_date(&captures);
+      if trigger_date.is_none() {
+        snooze_send_error_message(&channel, message, &ctx.http).await;
+      }
+      {
+        let mut db_instance = INSTANCE.write().unwrap();
+        db_instance.event_add(NewEvent {
+          author: snooze.author as i64,
+          channel: snooze.channel as i64,
+          content: &snooze.content,
+          trigger_date: trigger_date.unwrap().naive_utc(),
+        });
+      }
+      channel
+        .delete_message(&ctx.http, snooze.message_id)
+        .await
+        .unwrap();
+      message.react(&ctx.http, '✅').await.unwrap();
+      let mut list_snooze = SNOOZE.lock().unwrap();
+      list_snooze.retain(|s| s.message_id != snooze.message_id);
+      let list_snooze_message = SNOOZE_MESSAGE.lock().unwrap();
+      list_snooze_message.iter().cloned().for_each(|m| {
+        let http = ctx.http.clone();
+        if m.channel_id == channel {
+          tokio::spawn(async move {
+            channel.delete_message(http, m.id.0).await.unwrap();
+          });
+        }
+      });
```

Comment:
To avoid cloning and having to tokio spawn
You can first find the ids you want to delete first an then call delete on them

Reply:
You are removing all snooze messsages ?
What if 2 users are snoozing, the one that finish first is deleting the other "users" messages ?

Reply:
Also you never delete anything that is in `SNOOZE_MESSAGE` 
so ... you are probably re calling delete_message on something already delete